lib/tasks/update-package-json.js

```diff
@@ -71,23 +75,57 @@ function removePackages(packageJSON) {
   }
 }
 
+const states = {
+  updated: 1,
+  added: 2,
+};
+
 function addPackages(packageJSON) {
   console.log('Adding new required dependencies.');
-  let wasDepChange = false;
+  let hasNewDevDep = false;
   for (const [dep, version] of packagesToAdd) {
-    if (packageJSON['dependencies'] && packageJSON['dependencies'][dep]) {
-      wasDepChange = true;
-      packageJSON['dependencies'][dep] = version;
-    } else {
-      packageJSON['devDependencies'][dep] = version;
+    if (
+      updateVersion(packageJSON['dependencies'], dep, version) ===
+      states.updated
+    ) {
+      continue;
+    }
+    if (
+      updateVersion(packageJSON['devDependencies'], dep, version) ===
+      states.added
+    ) {
+      hasNewDevDep = true;
     }
   }
-  if (wasDepChange) {
-    packageJSON['dependencies'] = sortDependencies(packageJSON['dependencies']);
+  if (hasNewDevDep) {
+    packageJSON['devDependencies'] = sortDependencies(
+      packageJSON['devDependencies'],
+    );
   }
-  packageJSON['devDependencies'] = sortDependencies(
-    packageJSON['devDependencies'],
-  );
+}
+
+function updateVersion(deps, depToAdd, minimumVersion) {
+  if (!deps) return;
+  if (deps[depToAdd]) {
+    let version = deps[depToAdd];
+
+    // Remove the ~ ^ symbols to compare correctly when they are different
+    let trimedVersion = isNaN(version[0]) ? version.slice(1) : version;
```

Comment:
we should use the `semver` library to do this comparison I think `semver.gte` is what we're looking for 👍

Reply:
that's weird, it didn't pop in my search results earlier 🤔

Reply:
oh, but it looks like it doesn't support `^` and `~` in the compare functions 🤔

Reply:
ok I found how this thing works nevermind.